src/test/java/org/jenkinsci/plugins/plaincredentials/FileCredentialsTest.java

```diff
@@ -0,0 +1,110 @@
+package org.jenkinsci.plugins.plaincredentials;
+
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import org.apache.commons.fileupload.FileItem;
+import org.apache.commons.fileupload.FileItemHeaders;
+import org.jenkinsci.plugins.plaincredentials.impl.FileCredentialsImpl;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+
+public class FileCredentialsTest {
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldThrowAnExceptionIfFileNameIsBlank() throws IOException {
+        FileCredentials fileCredentials = new FileCredentialsImpl(CredentialsScope.GLOBAL, "1", "", new StubFileItem(), "", "");
+    }
+
+    private class StubFileItem implements FileItem {
```

Comment:
Yes it is a possibility but I used this approach to avoid having to set all the parameters in the constructor.